python/cudf/cudf/core/join/join.py

```diff
@@ -307,34 +307,72 @@ def perform_merge(self) -> cudf.DataFrame:
             left_key.set(self.lhs, lcol_casted)
             right_key.set(self.rhs, rcol_casted)
 
-        left_rows, right_rows = self._gather_maps(
-            left_join_cols, right_join_cols
-        )
-        gather_kwargs = {
-            "keep_index": self._using_left_index or self._using_right_index,
-        }
-        left_result = (
-            self.lhs._gather(
-                GatherMap.from_column_unchecked(
-                    left_rows, len(self.lhs), nullify=True
+        if self.how == "cross":
+            lib_table = plc.join.cross_join(
+                plc.Table(
+                    [
+                        col.to_pylibcudf(mode="read")
+                        for col in self.lhs._data.columns
+                    ]
                 ),
-                **gather_kwargs,
-            )
-            if left_rows is not None
-            else cudf.DataFrame._from_data({})
-        )
-        del left_rows
-        right_result = (
-            self.rhs._gather(
-                GatherMap.from_column_unchecked(
-                    right_rows, len(self.rhs), nullify=True
+                plc.Table(
+                    [
+                        col.to_pylibcudf(mode="read")
+                        for col in self.rhs._data.columns
+                    ]
                 ),
-                **gather_kwargs,
             )
-            if right_rows is not None
-            else cudf.DataFrame._from_data({})
-        )
-        del right_rows
+            columns = lib_table.columns()
+            left_names, right_names = (
+                self.lhs._data.names,
+                self.rhs._data.names,
+            )
+            left_result = cudf.DataFrame._from_data(
+                {
+                    col: ColumnBase.from_pylibcudf(lib_col)
+                    for col, lib_col in zip(
+                        left_names, columns[: len(left_names)]
+                    )
+                }
+            )
+            right_result = cudf.DataFrame._from_data(
+                {
+                    col: ColumnBase.from_pylibcudf(lib_col)
+                    for col, lib_col in zip(
```

Comment:
Could you include `strict=True` in this `zip` call?

Reply:
Done 👍

Reply:
While we're in the spirit of adding Ruff rules, let's do this one: https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/

Reply:
Yup definitely. Looks like we have an open issue to implement this eventually https://github.com/rapidsai/cudf/issues/15835